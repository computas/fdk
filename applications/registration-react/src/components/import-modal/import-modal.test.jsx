import React from 'react';
import { shallow } from 'enzyme';
import AppImportModal from './import-modal.component';

let defaultProps;
let wrapper;
let handleAction;
let toggle;
let preventDefault;

beforeEach(() => {
  handleAction = jest.fn();
  toggle = jest.fn();
  preventDefault = jest.fn();
  defaultProps = {
    handleAction,
    toggle,
    modal: true,
    className: null,
    title: 'Tittel',
    body: 'body',
    setURL: 'http://www.google.no',
    btnConfirm: 'btnConfirm',
    btnCancel: 'btnCancel'
  };
  wrapper = shallow(<AppImportModal {...defaultProps} />);
});

test('should render AppImportModal correctly', () => {
  expect(wrapper).toMatchSnapshot();
});

test('should handle handleAction', () => {
  const mockedEvent = {
    preventDefault
  };
  wrapper.setState({
    url: 'url'
  });
  wrapper
    .find('Button')
    .first()
    .prop('onClick')(mockedEvent);
  expect(handleAction).toHaveBeenCalled();
});

test('should handle handleAction', () => {
  const mockedEvent = {
    preventDefault
  };
  wrapper
    .find('Button')
    .first()
    .prop('onClick')(mockedEvent);
  expect(handleAction).not.toHaveBeenCalled();
});

test('should handle onBtnCancel', () => {
  const mockedEvent = {
    preventDefault
  };
  wrapper
    .find('Button')
    .at(1)
    .prop('onClick')(mockedEvent);
  expect(toggle).toHaveBeenCalled();
});
