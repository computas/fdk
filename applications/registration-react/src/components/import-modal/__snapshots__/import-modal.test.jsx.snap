// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render AppImportModal correctly 1`] = `
<div>
  <Modal
    autoFocus={true}
    backdrop={true}
    backdropTransition={
      Object {
        "mountOnEnter": true,
        "timeout": 150,
      }
    }
    centered={false}
    className={null}
    fade={true}
    isOpen={true}
    keyboard={true}
    modalTransition={
      Object {
        "timeout": 300,
      }
    }
    onClosed={[Function]}
    onOpened={[Function]}
    role="dialog"
    toggle={[Function]}
    zIndex={1050}
  >
    <ModalHeader
      closeAriaLabel="Close"
      tag="h5"
      toggle={[MockFunction]}
      wrapTag="div"
    >
      Tittel
    </ModalHeader>
    <ModalBody
      tag="div"
    >
      body
      <div
        className="d-flex flex-column align-items-center mt-5"
      >
        <label
          className="fdk-form-label w-100"
          htmlFor="importURL"
        >
          http://www.google.no
          <input
            className="form-control"
            id="importURL"
            onChange={[Function]}
            type="input"
          />
        </label>
      </div>
    </ModalBody>
    <ModalFooter
      className="d-flex justify-content-center"
      tag="div"
    >
      <Button
        color="danger"
        onClick={[Function]}
        tag="button"
      >
        btnConfirm
      </Button>
      <Button
        color="primary"
        onClick={[Function]}
        tag="button"
      >
        btnCancel
      </Button>
    </ModalFooter>
    <ModalFooter
      tag="div"
    />
  </Modal>
</div>
`;
